synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -109,6 +111,23 @@ def _get_rules_for_room(self, room_id):
         )
 
     @defer.inlineCallbacks
+    def _get_sender_power_level(self, event, context):
+        pl_event_key = (EventTypes.PowerLevels, "", )
+        if pl_event_key in context.prev_state_ids:
+            # fastpath: if there's a power level event, that's all we need, and
+            # not having a power level event is an extreme edge case
+            auth_events_ids = [context.prev_state_ids[pl_event_key]]
+        else:
+            auth_events_ids = yield self.auth.compute_auth_events(
+                event, context.prev_state_ids, for_verification=False,
+            )
+        auth_events = yield self.store.get_events(auth_events_ids)
+        auth_events = {
+            (e.type, e.state_key): e for e in auth_events.values()
```

Comment:
you want `.itervalues()` here

Reply:
I'd probably also move this in to the else, and explicitly do:

```python
pl_event_key = (EventTypes.PowerLevels, "", )
pl_event_id = context.prev_state_ids.get((EventTypes.PowerLevels, "",))
if pl_event_id:
    pl_event = yield self.store.get_event(pl_event_id)
    auth_events = { pl_event_key: pl_event }
else:
    ...
defer.returnValue(get_user_power_level(event.sender, auth_events))
```

to avoid some allocations

Reply:
is it worth s/values/itervalues/ in the place I copied it from and all the other places that copied it from that too? :)

Reply:
Almost certainly, so long as its obvious how its used